auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -512,6 +575,50 @@ interface CredentialParseFunction {
         void parseCredential(String algorithm, String format, String body) throws RealmUnavailableException, XMLStreamException;
     }
 
+    /**
+     * Re-generate the signatures for all the identities in this realm.
+     * This method is intended to be called after updating the key pair used by this realm.
+     *
+     * @throws RealmUnavailableException Thrown if unable to obtain the realm identity iterator or dispose and close the realm
```

Comment:
I think we could reword this to something like:

@throws RealmUnavailableException if the realm is not able to handle requests for any reason

Reply:
Updated